[compile error] Fix openNURBS compile issue under Windows

Currently you are getting following compile issue under windows in case you have `BUILD_surface_on_nurbs` enabled:

```
example_nurbs_viewer_surface.obj : error LNK2019: unresolved external symbol ON_ErrorEx referenced in function "public: class ONX_Model_Object & __cdecl ON_ClassArray<class ONX_Model_Object>::operator[](int)" (??A?$ON_ClassArray@VONX_Model_Object@@@@QEAAAEAVONX_Model_Object@@H@Z)
35>
```

Reason is the missing macro `ON_DLL_TEMPLATE`, caused by missing macro `ON_DLL_EXPORTS`

https://github.com/PointCloudLibrary/pcl/blob/8eb47684da26d42cd8cd69c79976e8aee04a2831/surface/include/pcl/surface/3rdparty/opennurbs/opennurbs_extensions.h#L139-L167

The macro will be defined here:
https://github.com/PointCloudLibrary/pcl/blob/c44061819582ac6a641b20c1bbd3b44c31deb813/surface/include/pcl/surface/3rdparty/opennurbs/opennurbs_defines.h#L88-L123

To enable this block `OPENNURBS_EXPORTS` needs to be defined
https://github.com/PointCloudLibrary/pcl/blob/c44061819582ac6a641b20c1bbd3b44c31deb813/surface/include/pcl/surface/3rdparty/opennurbs/opennurbs_defines.h#L66-L70

And surprise: `OPENNURBS_EXPORTS` will be never be defined.

A long time ago there was already a [try to fix](20835db9296735b4fb83f13c0477fbd116da109a) issues with it, but this fix was incomplete.

So we have the choice:
* Adding `OPENNURBS_EXPORTS` flag: In this case we need to export `OPENNURBS_IMPORTS`, too and revert [the old fix](20835db9296735b4fb83f13c0477fbd116da109a). Otherwise you are getting tons of warnings: `warning C4141: 'dllexport': used more than once (compiling source file C:\dev\pcl\surface\src\3rdparty\opennurbs\opennurbs_3dm_attributes.cpp)`
* Replacing `ON_CLASS` and all other openNURBS specific macros by e.g. `PCL_EXPORTS`. In this case we have currently no replacement for e.g. `ON_DLL_TEMPLATE`

--- a/surface/include/pcl/surface/3rdparty/opennurbs/opennurbs_error.h
+++ b/surface/include/pcl/surface/3rdparty/opennurbs/opennurbs_error.h
@@ -86,7 +86,7 @@ void    ON_Error( const char*, /* sFileName:   __FILE__ will do fine */
                   ...          /* printf() style ags */
                   );
 
-ON_DECL
+PCL_EXPORTS ON_DECL
 void    ON_ErrorEx( const char*, // sFileName:   __FILE__ will do fine
                   int,           // line number: __LINE__ will do fine
                   const char*,   // sFunctionName: __FUNCTION__ will do fine

